### PR TITLE
HPCC-18030 Simplify ITimeReporter and refactor inefficient code

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
Here as a reminder - needs rebasing once the commit it is dependent on is merged.

**Comment by ghalliday** [comment]
@richardkchapman please review.

This is from a while back.  I think it is deleting redundant code, and cleaning up the code.  The change to using the hash table iterator means something is O(N) instead O(N^2).

File: system/jlib/jdebug.hpp [hunk 1]
```diff
@@ -123,28 +123,20 @@ struct UserSystemTime_t
...
-  virtual __int64 getMaxTime(unsigned idx) = 0;
-  virtual unsigned getCount(unsigned idx) = 0;
-  virtual StatisticKind getTimerType(unsigned idx) = 0;
-  virtual StatisticScopeType getScopeType(unsigned idx) = 0;
-  virtual StringBuffer &getScope(unsigned idx, StringBuffer &s) = 0;
   virtual StringBuffer &getTimings(StringBuffer &s) = 0;
   virtual void printTimings() = 0;
   virtual void reset() = 0;
-  virtual void mergeInto(ITimeReporter &other) = 0;
+  virtual void mergeInto(ITimeReporter &other) = 0; // Could be removed - not actually used.
```
Comment (richardkchapman) [inline system/jlib/jdebug.hpp]: Perhaps this would be a good point to do so?

**Comment by richardkchapman** [comment]
@ghalliday One question

File: system/jlib/jdebug.hpp (hunk 1 above)
Comment (ghalliday) [reply system/jlib/jdebug.hpp]: Actually the comment isn't right - it is used internally within the implementation.  I will change the comment.

**Comment by ghalliday** [comment]
@richardkchapman comment was wrong - see response above.  I think ready to merge.

**Pull request merged**